fhir-persistence-schema/docs/DB2SchemaMigration.md

```diff
@@ -0,0 +1,382 @@
+
+The variable privileges (grants) are updated one time, and applied to `FHIRUSER` group. 
+
+## Managing TABLE
+There are three administrative tables in `FHIR_ADMIN`. These table definitions are more completely described in [DB2MultiTenancy.md](DB2MultiTenancy.md). 
+
+**Table: VERSION_HISTORY**
+
+Before any table or schema object is created, the `VERSION_HISTORY` table is created using `CreateVersionHistory.createTableIfNeeded(adminSchemaName, adapter);`  This table includes VERSION_HISTORY for resources created in the schema. If an object in the schema is updated, then the VERSION_HISTORY table must be updated to track the changes. 
+
+The `VERSION_HISTORY` table enables Tenant version isolation, so upgrades and patches are separately applied to the schema the tenant belongs. Specifically, if the tenant shares the table definitions with another client, the patching is applied to both tenants at the same time.  If the tenant has a specific schema, the table definitions may be updated independent of each other. 
+
+The `CreateVersionHistory` class controls the creation the `VERSION_HISTORY` table. Importantly, the VERISON_HISTORY table does not support migrations and changes. The Primary Key has a corresponding index `PK_VERSION_HISTORY`, and the code does not support updating the index. 
+
+This table is an administrative table and should not require updating and migration. 
+
+**Table: TENANTS**
+
+The `FHIR_ADMIN.TENANTS` table maps a given tenant to the Multi-Tenant ID (MT_ID) for the given tenant.  The MT_ID is subsequently used to restrict row access based on the matching value. 
+
+`AdminSchemaGenerator.addTenantTable` method adds the TENANT table, and updates to this table definition requries a change to the setVersion (incrementing by 1). A corresponding change must be made to `FHIRSchemaGenerator.addTenantTable` method.
+
+``` java
+this.tenantsTable = Table.builder(adminSchemaName, TENANTS)
+    .setVersion(1)
+    .addIntColumn(            MT_ID,             false)
+    .addVarcharColumn(      TENANT_NAME,        36,  false)
+    .addVarcharColumn(    TENANT_STATUS,        16,  false)
+    .addUniqueIndex(IDX + "TENANT_TN", TENANT_NAME)
+    .addPrimaryKey("TENANT_PK", MT_ID)
+    .setTablespace(fhirTablespace)
+    .build(model);
+```
+
+If this table is updated, the changes must be manually apply the migration steps to the `TENANTS` and the corresponding indices. 
+
+**Table: TENANT_KEYS**
+
+The Tenant Keys table stores a hashed version of the tenant specific key. Upon creating a session and calling the SET_TENANT procedure, the SET_TENANT queries the incoming key against the TENANT_KEYS database to map to a specific tenant MT_ID. 
+
+`AdminSchemaGenerator.addTenantKeysTable` method adds the TENANT_KEYS table, and updates to this table definition requries a change to the setVersion (incrementing by 1).
```

Comment:
should the example just below this show the `setVersion` thing?  if not, maybe change the order so that the snippet comes before the comment about `setVersion`.

Reply:
I forgot to add it! - added inline